frame/bounties/src/lib.rs

```diff
@@ -653,12 +786,656 @@ decl_module! {
+				// Extend expiry for active subbounty
+				if bounty.activesubbounty.len() > 0 {
+					bounty.activesubbounty
+					.iter()
+					.for_each(|subbounty_id| {
+						let _ = Self::impl_extend_subbounty_expiry(&signer, bounty_id, *subbounty_id, remark.clone());
+					});
+				}
 				Ok(())
 			})?;
 
 			Self::deposit_event(Event::<T>::BountyExtended(bounty_id));
 		}
+
+		/// Add a new subbounty.
+		///
+		/// The dispatch origin for this call must be curator.
+		/// Bounty must me in "active" state.
+		///
+		/// Subbouty gets added successfully & fund gets reserved, if bounty has enough fund.
+		/// else call get failed.
+		///
+		/// Upperbount to maximum number of subbounties that can be added is
+		/// managed via runtime trait config 'MaxSubBountyCount'.
+		///
+		/// Payment: `TipReportDepositBase` will be reserved from the origin account, as well as
+		/// `DataDepositPerByte` for each byte in `reason`. It will be unreserved upon approval,
+		/// or slashed when rejected.
+		///
+		/// if call is success, state of subbounty is moved to "Approved" state.
+		/// And later moved to "Funded" state as part of "spend_fund()" callback.
+		///
+		/// - `bounty_id`: Bounty ID for which subbounty to be added.
+		/// - `value`: Value for executing the proposal.
+		/// - `description`: Text description for the subbounty.
+		#[weight = 10_000]
+		fn add_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			value: BalanceOf<T>,
+			description: Vec<u8>,
+		)
+		{
+			let signer = ensure_signed(origin)?;
+
+			// TODO :: Have to recheck
+			// Governance design choice, which is more appropriate for the context ?
+			//
+			// * add_subbounty() :: right now origin must me signed
+			// & should be master curator. and state of subbounty
+			// is moved to `Approved` state.
+			//
+			// But logic can be refactored, similar to parent bounty sequence flow...
+			//
+			// * propose_subbounty() :: origin may be signed, bond deposit is
+			// applicable & state of subbonty is moved to `Proposed` state.
+			//
+			// * approve_subbounty() :: origin is signed & should be master-curator,
+			// & move the state of subounty to `Approved`.
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+				match bounty.status {
+					BountyStatus::Active {
+						ref curator,
+						..
+					} => {
+						ensure!(signer == *curator, Error::<T>::RequireCurator);
+						// Verify the arguments
+						ensure!(description.len() <=
+							T::MaximumReasonLength::get() as usize,
+							Error::<T>::ReasonTooBig,
+						);
+						ensure!(value >= T::BountyValueMinimum::get(),
+							Error::<T>::InvalidValue,
+						);
+						ensure!((bounty.activesubbounty.len() as u32) <
+							T::MaxSubBountyCount::get() as u32,
+							Error::<T>::SubBountyMaxOverflow,
+						);
+						// reserve deposit for new subbounty
+						let bond = T::BountyDepositBase::get()
```

Comment:
no need to reserve deposit

Reply:
As suggested, design for subbounty fund management got changed, Now each subbounty got unique account.
fund reservation logic is removed.